### PR TITLE
Remove dependency on JNR and Truffle

### PR DESCRIPTION
**Pull request by pitr-ch**

### CODE REVIEW

File: src/main/java/org/truffleruby/platform/Platform.java [hunk 1]
```diff
@@ -287,18 +287,26 @@ private static boolean startsWithIgnoreCase(String s1, String s2) {
...
+        String osNameProperty = System.getProperty("os.name");
+        final Map<String, String> jnrOsNames = new HashMap<String, String>();
+        jnrOsNames.put("Mac OS X", "darwin");
+        jnrOsNames.put("Darwin", "darwin");
+        jnrOsNames.put("Linux", "linux");
 
-        return theOSName == null ? OSName : theOSName;
+        String osName = jnrOsNames.getOrDefault(osNameProperty, osNameProperty);
+        return RUBY_OS_NAMES.getOrDefault(osName, osName);
     }
```
Comment (chrisseaton) [inline src/main/java/org/truffleruby/platform/Platform.java]: We need to better preserve the licenses of `jar-posix` here. Maybe put the methods you're using into a separate file and include the license text there?

Comment (chrisseaton) [inline src/main/java/org/truffleruby/platform/Platform.java]: Or change the comments if it isn't really a case of inlining code from the other project, and is actually a reimplementation.

File: src/main/java/org/truffleruby/platform/Platform.java [hunk 2]
```diff
@@ -286,19 +286,32 @@ private static boolean startsWithIgnoreCase(String s1, String s2) {
             || s1.toLowerCase(java.util.Locale.ENGLISH).startsWith(s2.toLowerCase(java.util.Locale.ENGLISH));
     }
 
+    // compatible with jnr.posix.util.Platform.getOSName()
+    private static String getJNRCompatibleOSName() {
+        String osNameProperty = System.getProperty("os.name");
+        final Map<String, String> jnrOsNames = new HashMap<>();
+        jnrOsNames.put("Mac OS X", "darwin");
+        jnrOsNames.put("Darwin", "darwin");
+        jnrOsNames.put("Linux", "linux");
```
Comment (eregon) [inline src/main/java/org/truffleruby/platform/Platform.java]: Should this be a static Map or just a switch?

Comment (pitr-ch) [inline src/main/java/org/truffleruby/platform/Platform.java]: yeah switch is better.

File: src/main/java/org/truffleruby/Main.java [hunk 3]
```diff
@@ -61,6 +60,9 @@
 
 public class Main {
 
+    public static final boolean IS_AOT = Boolean.logicalOr(
+            Boolean.getBoolean("com.oracle.graalvm.isaot"),
+            Boolean.getBoolean("com.oracle.truffle.aot"));
```
Comment (eregon) [inline src/main/java/org/truffleruby/Main.java]: Why not just `||`?

File: src/main/java/org/truffleruby/Main.java [hunk 4]
```diff
@@ -122,8 +124,9 @@ public static void main(String[] args) {
                     exitCode = status ? 0 : 1;
                 } else {
                     if (!isGraal() && config.getOption(OptionsCatalog.GRAAL_WARNING_UNLESS)) {
-                        Log.performanceOnce("this JVM does not have the Graal compiler - performance will be limited" +
-                                " - see doc/user/using-graalvm.md");
+                        LogWithoutTruffle.performanceOnce(
+                                "this JVM does not have the Graal compiler - performance will be limited" +
+                                        " - see doc/user/using-graalvm.md");
```
Comment (eregon) [inline src/main/java/org/truffleruby/Main.java]: Please avoid spreading into too many lines single statements.

File: src/main/java/org/truffleruby/Main.java (hunk 3 above)
Comment (pitr-ch) [inline src/main/java/org/truffleruby/Main.java]: Alignment.

Comment (eregon) [inline src/main/java/org/truffleruby/Main.java]: This is not worth the loss in readability and the need to read that many characters.

**Pull request merged**